app/services/process_submission_service.rb

```diff
@@ -24,106 +22,51 @@ def perform # rubocop:disable Metrics/AbcSize
       end
     end
-
-    submission.detail_objects.to_a.each do |submission_detail|
-      send_email(submission_detail) if submission_detail.instance_of? EmailSubmissionDetail
-    end
-
-    # explicit save! first, to save the responses
-    submission.save!
-    submission.complete!
   end
 
   private
 
-  def email_body_parts(email)
-    {
-      'text/plain' => email.email_body
-    }
-  end
-
-  def send_email(mail)
-    email_body = email_body_parts(mail)
+  attr_reader :submission, :payload_service
 
-    if number_of_attachments(mail) <= 1
-      response = EmailService.send_mail(
-        from: mail.from,
-        to: mail.to,
-        subject: mail.subject,
-        body_parts: email_body,
-        attachments: attachments(mail)
-      )
+  def generate_attachments(attachments_payload, token)
+    tmp_file_map = DownloadService.new(
+      attachments: attachments_payload,
+      target_dir: nil,
+      token: token
+    ).download_in_parallel
+    attachments = []
 
-      submission.responses << response.to_h
-    else
-      attachments(mail).each_with_index do |a, n|
-        response = EmailService.send_mail(
-          from: mail.from,
-          to: mail.to,
-          subject: "#{mail.subject} {#{submission_id}} [#{n + 1}/#{number_of_attachments(mail)}]",
-          body_parts: email_body,
-          attachments: [a]
-        )
-
-        submission.responses << response.to_h
-      end
+    tmp_file_map.each do |attachment_info|
+      attachment = Attachment.new(filename: attachment_info[:filename], mimetype: attachment_info[:mimetype])
+      attachment.path = attachment_info[:tmp_path]
+      attachments << attachment
     end
-  end
-  # rubocop:enable Metrics/MethodLength
 
-  def number_of_attachments(mail)
-    attachments(mail).size
-  end
-
-  # returns array of urls
-  # this is done over all files so we download all needed files at once
-  def unique_attachment_urls
-    attachments = submission.detail_objects.map(&:attachments).flatten
-    urls = attachments.map { |e| e['url'] }
-    urls.compact.sort.uniq
-  end
-
-  def submission
-    @submission ||= Submission.find(submission_id)
-  end
-
-  def headers
-    { 'x-encrypted-user-id-and-token' => submission.encrypted_user_id_and_token }
-  end
-
-  def attachments(mail)
-    attachments = mail.attachments.map(&:with_indifferent_access)
-    attachment_objects = AttachmentParserService.new(attachments: attachments).execute
-
-    attachments.each_with_index do |value, index|
-      if value[:pdf_data]
-        attachment_objects[index].file = generate_pdf({ submission: value[:pdf_data] }, @submission_id)
-      else
-        attachment_objects[index].path = download_attachments[attachment_objects[index].url]
-      end
-    end
-    attachment_objects
-  end
-
-  def download_attachments
-    @download_attachments ||= DownloadService.download_in_parallel(
-      urls: unique_attachment_urls,
-      headers: headers
-    )
+    attachments
   end
```

Comment:
if the `DownloadService`, and similarly the `SaveTempPdf` below, could return an `Attachment` object, or an array of `Attachment` objects, then we could move this logic out to those services. I think it would be a simpler workflow as the Services simply return exactly what we need with no need for this intermediate step of creating `Attachment` objects and assigning tmp files to them.

Reply:
This would be the next step though